www/source/blog/2017-10-24-node-with-db.html.md

```diff
@@ -0,0 +1,348 @@
+---
+title: NodeJS Scaffolding with Databases!
+date: 2017-10-24
+author: Nell Shamrell-Harrington
+tags: Scaffolding, Node
+category: community
+classes: body-article
+---
+
+Greetings fellow Habicats! Good news today - you can use the Node scaffolding to Habitize Node applications that use a database!
+
+One of my favorite resources on Node is ["Express in Action" by Evan Hahn](https://www.manning.com/books/express-in-action). In one of the chapters of the book, you create an application called "Learn About Me" - a simple social networking application that uses a MEN stack (MongoDB, Express, and NodeJS).  Let's Habitize this application.
+
+## Pre-requisites
+* A Github Account
+* NodeJS installed on your workstation
+* Habitat installed and set up on your workstation
+* Docker installed and running on your workstation
+* Text editor of your choice (in this post I use vim, but feel free to substitute your preferred editor)
+
+## Setting up your application
+
+Go ahead and clone the source code for "Learn About Me" with this command:
+
+```console
+$ git clone git@github.com:EvanHahn/Express.js-in-Action-code.git
+$ cd Express.js-in-Action-code/Chapter_08/learn-about-me
+```
+
+Now install the application's dependencies
+
+```console
+$ npm install
+```
+
+If you'd like, go ahead and run the application locally:
+
+```console
+$ npm start
+```
+
+And you should be able to access the app at http://localhost:3000.
+
+## Habitizing your application - Part I
+
+```console
+$ hab plan init -s node
+```
+
+There are two parts to this application - the application itself and the database.  We'll run these in two separate containers, but the application needs to be able to access and talk to the database.
+
+Let's set up the default configuration for the application's connection to the database in habitat/default.toml - this file sets environmental variables that will be used by the application.
```

Comment:
What's the rule here on paths? Surround them with `habitat/default.toml` or let them ride?

Reply:
"this file sets" makes me think that this file is going to do the setting of the environment variables. Also these values are never used. Later you don't mention that you could connect now with the default values found in `habitat/default.toml` file within your config using:

```
# ~/learn-about-me/habitat/config/config.json
{
  "mongo": {
      "host" : "{{cfg.mongo.host}}",
      "port"   : "{{cfg.mongo.port}}"
    }
}
```

Reply:
Good point - fixing!